src/v/datalake/translation/tests/partition_translator_tests.cc

```diff
@@ -359,12 +427,48 @@ TEST_F_CORO(partition_translator_fixture, test_coordinator_retries) {
     // simulate no leader conditions
     test_ctx.set_error_on_add_translated_files(true);
     co_await add_translator(test_ctx);
-    co_await test_ctx.wait_for_translation_attempts(10);
     // Ensure translation does not make any progress
     ASSERT_LT_CORO(test_ctx.max_translated_offset(), kafka::offset{0});
     test_ctx.set_error_on_add_translated_files(false);
     co_await test_ctx.wait_for_translation_attempts(10);
     ASSERT_GT_CORO(test_ctx.max_translated_offset(), kafka::offset{0});
 }
 
-// todo: add a test for batching once the functionality is checked in
+TEST_F_CORO(partition_translator_fixture, test_batching) {
+    auto& test_ctx = make_test_context();
+    // Let the translator batch to the limit
+    test_ctx.set_should_finish_inflight_translation(false);
+    co_await add_translator(test_ctx);
+
+    auto stop = false;
+    auto produce_f = ss::do_until(
+      [&stop] { return stop; },
+      [&test_ctx]() {
+          auto current = test_ctx.max_translatable_offset();
+          test_ctx.set_max_translatable_offset(
+            current + kafka::offset_delta(10000));
+          return ss::sleep(1ms);
+      });
+
+    std::exception_ptr ex = nullptr;
+    try {
+        RPTEST_REQUIRE_EVENTUALLY_CORO(10s, [&test_ctx]() {
+            const auto& files = test_ctx.translated_files();
+            const auto it = files.find(test_ctx.ntp().tp);
+            if (it != files.end()) {
+                vlog(test_logger.trace, "translated files: {}", it->second);
+            }
+            return it != files.end() && it->second.size() >= 2
+                   && std::ranges::all_of(it->second, [](const auto& entry) {
+                          return entry.file_size_bytes >= 32_MiB;
+                      });
+        });
+    } catch (...) {
+        ex = std::current_exception();
+    }
+    stop = true;
+    co_await std::move(produce_f);
+    if (ex) {
+        std::rethrow_exception(ex);
```

Comment:
nit: RPTEST_FAIL_CORO(fmt::to_string(ex))`? Not sure about gtest but i know debugging exceptions thrown in test bodies with boost test is really annoying if you aren't the author of the test (and maybe still even if you are!)